test/integration-tests/IntegrationTests.Http/HttpTests.cs

```diff
@@ -0,0 +1,83 @@
+// <copyright file="HttpTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Linq;
+using FluentAssertions.Execution;
+using IntegrationTests.Helpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IntegrationTests.Http
+{
+    public class HttpTests : TestHelper
+    {
+        private const string ServiceName = "Samples.Http";
+
+        public HttpTests(ITestOutputHelper output)
+            : base("Http", output)
+        {
+            SetEnvironmentVariable("OTEL_SERVICE_NAME", ServiceName);
+            SetEnvironmentVariable("OTEL_DOTNET_AUTO_ENABLED_INSTRUMENTATIONS", "HttpClient,AspNet");
+        }
+
+        [Fact]
+        [Trait("Category", "EndToEnd")]
+        public void SubmitTraces()
+        {
+            var agentPort = TcpPortProvider.GetOpenPort();
+            using var agent = new MockZipkinCollector(Output, agentPort);
+
+            const int expectedSpanCount = 3;
+
+            using var processResult = RunSampleAndWaitForExit(agent.Port, enableStartupHook: true);
+            Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode} and exception: {processResult.StandardError}");
+            var spans = agent.WaitForSpans(expectedSpanCount, 500);
```

Comment:
I see the same 500 milliseconds in other tests but it seems rather optimistic to me, perhaps 3 seconds?